hw06-solid/src/main/java/ru/otus/service/WalletService.java

```diff
@@ -0,0 +1,21 @@
+package ru.otus.service;
+
+import ru.otus.enums.Banknote;
+
+import java.util.List;
+import java.util.Map;
+
+public interface WalletService {
+
+    void put(Banknote banknote, int count);
+
+    void put (Map<Banknote, Integer> map);
+
+    int getBalance();
+
+    Map<Banknote, Integer> getBanknotes(Map<Banknote, Integer> banknotes);
+
+    List<Banknote> getBanknotes();
```

Comment:
```suggestion
    Iterable<Banknote> getBanknotes();
```
Здесь достаточно `Iterable` и дальше, где этот метод используется.
`List` говорит, что в него можно что-то добавить, удалить.
Да, вы возвращаете копии - проблем не будет, но `Itarable` выглядит лучше.